broadinstitute/single_cell_portal_core
# Add associated raw counts dropdown to file upload (SCP-3657)

SCP-3657 Adds the raw count association dropdown to expression file upload

![image](https://user-images.githubusercontent.com/2800795/135520250-b12dba92-2bfe-4666-9856-493a7b22d154.png)

This also adds additional robustness to the way we assemble form payloads to send to Rails.  A future consideration might be whether we should add a save endpoint that accepts JSON, to avoid this html form conversion.  for now, keeping this endpoint makes sense as it preserves manage_study compatibility, and also FormData is how our file upload infrastructure expects files.

TO TEST:
  1. load a study in the upload wizard (with react upload wizard feature flag on)
  2. upload a raw counts file
  3. go to the processed file tab
  4. fill out a new expression file, and specify a raw counts file associated with it
  5. save it, and confirm it saves
  6. deselect the associated raw counts file 
  7. save again, and confirm the deselection is preserved
